ext/testing/mod.js

```diff
@@ -3,55 +3,33 @@ import {
 	op_diff_str,
 	op_runtime_state,
 	op_set_promise_sanitization_hook,
-	op_set_promise_sanitization_name,
-	op_test_sanitization,
+	op_set_promise_sanitized_test_name,
+	op_get_outstanding_ops,
+	op_get_pending_promises,
 } from "ext:core/ops";
 
 import { Printer } from "ext:sable/console/printer.js";
 import { styles } from "ext:sable/utils/ansi.js";
-import { textWidth } from "ext:sable/utils/text_width.js";
 
-/**
- * Error which gets thrown whenever:
- *  - TestContext is leaking async ops
- *  - Test gets started when async ops are still pending
- */
-class TestContextLeakingAsyncOpsError extends Error {
+class TestContextLeakingPendingPromisesError extends Error {
 	/**
-	 * @param {TestContext=} testContext
-	 * @param {boolean=} isAsync - whether given testContext callback returned a promise
+	 * @param {TestContext} testContext
+	 * @param {number} pendingPromises
+	 * @param {number} promiseLeniency
 	 */
-	constructor(testContext, isAsync) {
-		// TODO(Im-Beast): Replace this with pretty errors after they happen
-		let message = `
-You wanted to create a test, but there are still asynchronous ops running.
-Please make sure they've completed before you create the test.`;
-
-		if (testContext) {
-			message = `
-At least one asynchronous operation or a Promise was started in ${testContext.title} but never completed!
-Please await all your promises or resolve test promise when every asynchronous operation has finished:`;
-
-			if (isAsync) {
-				message += `
-test('${testContext.title}', async (ctx) => {
-  ...
---^^^ ops or Promises leak somewhere around here, are you sure you awaited every promise?
-});`;
-			} else {
-				const ptd = "-".repeat(textWidth(testContext.title));
-
-				message += `
-test('${testContext.title}', (ctx) => {
---------${ptd}^ this test is not asynchronous, but leaks asynchronous ops or Promises
-  ...
---^^^ ops or Promises leak somewhere around here, are you sure this test was meant to be synchronous?
-});`;
-			}
-		}
-
-		super(message);
+	constructor(testContext, pendingPromises, promiseLeniency) {
```

Comment:
I removed advanced error messages because they are pretty annoying to work with.
They will come back after I implement prettier error messages though.